core/src/main/java/hudson/model/Computer.java

```diff
@@ -176,6 +176,21 @@
 
     protected final Object statusChangeLock = new Object();
 
+    private transient final List<RuntimeException> terminatedBy = Collections.synchronizedList(new ArrayList
+            <RuntimeException>());
+
+    public void recordTermination() {
+        try {
+            throw new RuntimeException(String.format("Termination requested by %s", Thread.currentThread()));
+        } catch (RuntimeException e) {
+            terminatedBy.add(e);
+        }
+    }
+
+    public List<RuntimeException> getTerminatedBy() {
```

Comment:
(or can use `@Restricted(NoExternalUse.class)` if this is not intended to be a public API)